packages/components/src/components/floor-selector/floor-selector.tsx

```diff
@@ -0,0 +1,237 @@
+            return;
+        }
+
+        if (this.floorListElement.scrollTop > 0) {
+            this.floorListElement.classList.add('mi-floor-selector__list--fade-top');
+        } else {
+            this.floorListElement.classList.remove('mi-floor-selector__list--fade-top');
+        }
+
+        if (this.floorListElement.scrollTop === maxScrollTop) {
+            this.floorListElement.classList.remove('mi-floor-selector__list--fade-bottom');
+        } else {
+            this.floorListElement.classList.add('mi-floor-selector__list--fade-bottom');
+        }
+    }
+
+    /**
+     * Sets the clicked floor active.
+     *
+     * @param {PointerEvent} event
+     */
+    private onSelectFloor(event: PointerEvent): void {
+        this.mapsindoors.setFloor((event.currentTarget as HTMLElement).getAttribute('data-floor'));
+        this.currentFloor = this.mapsindoors.getFloor().toString();
+        this.currentFloorElement = event.currentTarget as HTMLElement;
+    }
+
+    /**
+     * Called every time the component is connected to the DOM.
+     * It resets the floorlist and the active floor when the building changes.
+     */
+    connectedCallback(): void {
+        this.mapsindoors.addListener('building_changed', () => {
+            this.floors = [];
+            const building = this.mapsindoors.getBuilding();
+            if (building) {
+                if (!this.mapsindoors.getFloor()) {
+                    return;
+                }
+                this.currentFloor = this.mapsindoors.getFloor().toString();
+
+                Object.keys(building.floors)
+                    .sort((a, b): any => (b as any) - (a as any))
+                    .forEach(floor => {
+                        building.floors[floor].index = floor;
+                        this.floors.push(building.floors[floor]);
+                    });
+            }
+        });
+    }
+
+    /**
+     * Renders the floor selector.
+     *
+     * @returns {JSX.Element}
+     */
+    render(): JSX.Element {
+        return (
+            <Host>
+                <div
+                    class='mi-floor-selector'
+                    ref={(element): HTMLDivElement => this.floorSelectorElement = element as HTMLDivElement}>
+                    <button
+                        class={`mi-floor-selector__button mi-floor-selector__button--${this.floorSelectorClosed ? 'closed' : 'open'}`}
+                        onClick={(): void => this.onToggleFloorSelector()}>
+                    </button>
+
+                    <div
+                        class='mi-floor-selector__list'
+                        ref={(element): HTMLDivElement => this.floorListElement = element as HTMLDivElement}
+                        onScroll={(): void => this.onScrollStyle()}>
+                        {this.floors.map((floor) => (
+                            <button
+                                data-floor={floor.index}
```

Comment:
Can Stencil / JSX help us eliminating the need for the data attribute?

Reply:
@ammapspeople I am not sure how, we need the data attribute for styling the floors later on and checking which floor is selected. Did you have a specific solution in mind?

Reply:
Nothing specific, no. But using data attributes is a sign that it carries some legacy from "vanilla" JS into a framework, which should help us simplifying stuff like that.

For example, the onClick on the button can just call `this.onSelect` with `floor.index` as parameter instead of passing the click event, which is then used to read the data attribute from the button.

Reply:
We discussed this in a meeting today, and agreed that some usages of the data attribute can be changed, but not all.